src/decode-pppoe.c

```diff
@@ -97,11 +92,16 @@ int DecodePPPOEDiscovery(ThreadVars *tv, DecodeThreadVars *dtv, Packet *p,
         return TM_ECODE_OK;
     }
 
-    while (pppoedt < (PPPOEDiscoveryTag*) (pkt + (len - sizeof(PPPOEDiscoveryTag))) && pppoe_length >=4 && packet_length >=4)
-    {
 #ifdef DEBUG
+    /* parse any tags we have in the packet */
+
+    uint32_t tag_length = 0;
+    const uint8_t *pkt_pppoedt = pkt + PPPOE_DISCOVERY_HEADER_MIN_LEN;
+    while (pkt_pppoedt < pkt_pppoedt + (len - sizeof(PPPOEDiscoveryTag)) && pppoe_length >= 4 &&
```

Comment:
can this be rewritten to be more readable? Both this and the original code are basically unreadable IMHO